src/audio/dai.c

```diff
@@ -838,6 +839,7 @@ static int dai_prepare(struct comp_dev *dev)
 		comp_set_state(dev, COMP_TRIGGER_RESET);
 		return -EINVAL;
 	}
+	dd->chan->srcid = dai_get_srcid(dd->dai, dai->direction, dd->stream_id);
 
```

Comment:
We need to think about this. For non-8ulp platforms this function will crash. At least return 0 if src_id is not implemented.